tests/TestData/Providers/Reflection/ReflectionClassesTestDataProviders.php

```diff
@@ -0,0 +1,50 @@
+<?php
+declare(strict_types=1);
+
+namespace StubTests\TestData\Providers\Reflection;
+
+use Generator;
+use StubTests\Model\PHPClass;
+use StubTests\Model\PHPInterface;
+use StubTests\Model\StubProblemType;
+use StubTests\TestData\Providers\EntitiesFilter;
+use StubTests\TestData\Providers\ReflectionStubsSingleton;
+
+class ReflectionClassesTestDataProviders
+{
+    public static function allClassesProvider(): ?Generator
+    {
+        $allClassesAndInterfaces = ReflectionStubsSingleton::getReflectionStubs()->getClasses() +
+            ReflectionStubsSingleton::getReflectionStubs()->getInterfaces();
+        foreach (EntitiesFilter::getFiltered($allClassesAndInterfaces) as $class) {
```

Comment:
Can we iterate just over `$allClassesAndInterfaces` without the call of `getFiltered` since it should do nothing?

Reply:
By default `getFiltered` returns entities excluding those that have "STUB_IS_MISSED" or "HAS_DUPLICATION" known problems
```
if ($entity->hasMutedProblem(StubProblemType::STUB_IS_MISSED)) {
                $hasProblem = true;
            }
            if ($entity->hasMutedProblem(StubProblemType::HAS_DUPLICATION)) {
                $hasProblem = true;
            }
```
That was done to avoid the explicit passing of `StubProblemType::<PROBLEM>` everywhere because actually, we are not interested in entities that intentionally have muted problems